src/emoji.ts

```diff
@@ -2,7 +2,7 @@ import twemoji from 'twemoji';
 import { EmojiStyle } from './settings/data';
 
 const regex =
-  /(\p{Emoji}|\p{Emoji_Presentation}|\p{Emoji_Modifier}|\p{Emoji_Modifier_Base}|\p{Emoji_Component}|\p{Extended_Pictographic})/gu;
+  /(?!#)(\p{Emoji}|\p{Emoji_Presentation}|\p{Emoji_Modifier}|\p{Emoji_Modifier_Base}|\p{Emoji_Component}|\p{Extended_Pictographic})/gu;
```

Comment:
refined matcher to explicitly exclude the hash ("#") symbol

this was necessary because the "#" character is recognized as a base for key-cap emoji sequences, like #️⃣, and was causing inadvertent matching of markdown headers. The negative look-ahead (?!#) at the beginning of the pattern ensures that the regex skips over instances of the "#" character, while still matching other emoji-related Unicode properties